nvflare/app_opt/xgboost/data_loader.py

```diff
@@ -21,7 +21,7 @@
 
 class XGBDataLoader(ABC):
     @abstractmethod
-    def load_data(self, client_id) -> Tuple[xgb.core.DMatrix, xgb.core.DMatrix]:
+    def load_data(self, client_id: str = None, app_dir: str = None) -> Tuple[xgb.core.DMatrix, xgb.core.DMatrix]:
```

Comment:
Since we are modifying the "interface" method definition.
We need to also change the "implementation" of this class.
Which is the "HIGGSDataLoader" at tree-based examples and histogram-based example

Reply:
Right, although those dataloaders only need the client_id and not the app_dir, so I made the parameters optional, doesn't that work?

Reply:
I suggest make them all consistent in method signature, so whatever class that calls them can use same calling mechanism.

To be more specific, in https://github.com/NVIDIA/NVFlare/blob/main/examples/advanced/xgboost/tree-based/jobs/bagging_base/app/custom/higgs_data_loader.py#L44

Change 
`def load_data(self, client_id):`

to 

`def load_data(self, client_id: str, app_dir: Optional[str] = None):`

Note that type hint "str" can't have None, so the correct one should be Optional[str]

Reply:
Although inside HIGGSDataLoader class it is NOT using app_dir.
BUT whatever thing call these child classes of XGBDataLoader will/might call it like:
`data_loader.load_data(client_id, app_dir)`
If the HIGGSDataLoader method signature is NOT consistent.
Then this call will raise an Exception.